### ISSUE TITLE
[Timer] Missing public API for TIMER6 config (linked to synchronous task)

### ISSUE THREAD

**Issue by jalinei** [owntech-foundation/Core#32]
User can choose synchronous task to TIMER6 but has no way to configure timer6 from public API.

**Comment by jalinei** [comment]
Following a discussion with @Ayoub-Farah I understood that timer6 has not to be defined before the controlTask. The controlTask itself will initialize Timer6 and create the synchronous task accordingly. 

For now this is clearly unclear as the workflow is different between `HRTIM` clock source and `timer6` clock source. In `HRTIM` mode you do have to define the counter period before launching the synchronous task whereas with `timer6` it is dealt with internally. 

I suggest we clarify the enum, so that end-user is not bothered with hardware related naming
- replace `HRTIM` with `pwm`
- replace `timer6` with `pwm_independant`
Timer6 mention should remain explicit in documentation.

**Comment by jalinei** [comment]
linked with #42

**Comment by pierre-haessig** [comment]
> I suggest we clarify the enum, so that end-user is not bothered with hardware related naming
> 
>     * replace `HRTIM` with `pwm`
> 
>     * replace `timer6` with `pwm_independant`
>       Timer6 mention should remain explicit in documentation.

Good idea. My suggestion for the names would be to make them clearly opposite of each others:

* HRTIM: `pwm_synced`
* timer6: `pwm_unsynced` (or perhaps, independent, but then with an "e" instead of an "a")
